extensions/http/jersey-micrometer/src/main/java/org/eclipse/dataspaceconnector/extension/jersey/JerseyMicrometerExtension.java

```diff
@@ -0,0 +1,65 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.extension.jersey;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.binder.jersey.server.DefaultJerseyTagsProvider;
+import io.micrometer.core.instrument.binder.jersey.server.MetricsApplicationEventListener;
+import org.eclipse.dataspaceconnector.spi.EdcSetting;
+import org.eclipse.dataspaceconnector.spi.WebService;
+import org.eclipse.dataspaceconnector.spi.system.Inject;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtension;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtensionContext;
+
+/**
+ * An extension that register a Micrometer {@link MetricsApplicationEventListener} into Jersey to
+ * provide metrics and request timings.
+ */
+public class JerseyMicrometerExtension implements ServiceExtension {
+
+    @EdcSetting
+    public static final String ENABLE_METRICS = "edc.metrics.enabled";
+    @EdcSetting
+    public static final String ENABLE_JERSEY_METRICS = "edc.metrics.jersey.enabled";
+
+    @Inject
+    private WebService webService;
+
+    @Inject
+    private MeterRegistry meterRegistry;
+
+    @Override
+    public String name() {
+        return "Jersey Micrometer Metrics";
+    }
+
+    @Override
+    public void initialize(ServiceExtensionContext context) {
+        var enableMetrics = context.getSetting(ENABLE_METRICS, true);
+        var enableJerseyMetrics = context.getSetting(ENABLE_JERSEY_METRICS, true);
+
+        if (enableMetrics && enableJerseyMetrics) {
+            enableJerseyControllerMetrics();
+        }
+    }
+
+    private void enableJerseyControllerMetrics() {
+        webService.registerResource(new MetricsApplicationEventListener(
+                meterRegistry,
+                new DefaultJerseyTagsProvider(),
+                /* metricName = */ "jersey",
```

Comment:
Comments could be avoided extracting variables/constants with meaningful names

Reply:
Actually modern IDEs like IntelliJ automatically blend in param names. Removing comments
<img width="518" alt="Screenshot 2022-02-28 at 14 08 26" src="https://user-images.githubusercontent.com/5342234/155988654-e6df1adc-95eb-4297-a845-05a3abda43e2.png">
.